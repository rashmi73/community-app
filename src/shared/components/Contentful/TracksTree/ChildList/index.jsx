/**
 * The child list rendering.
 */
import _ from 'lodash';
import PT from 'prop-types';
import React from 'react';
import { themr } from 'react-css-super-themr';

import ChildListRow from '../ChildListRow';
import defaultTheme from './themes/default.scss';


export function ChildListInner(props) {
  const {
    theme,
    list,
    onSelected,
  } = props;

  return (
    <div className={theme['child-list']}>
      {
        _.map(list, option => (
          <ChildListRow
            onSelected={onSelected}
            key={option.id}
            list={option}
            selected={!!option.selected}
          >
            {(option.items.length > 0) && (
              <ChildListInner
                list={option.items}
                theme={theme}
                onSelected={onSelected}
              />
            )}
          </ChildListRow>
        ))
      }
    </div>
  );
}

ChildListInner.defaultProps = {
  onSelected: () => {},
};

ChildListInner.propTypes = {
  theme: PT.shape({
    'child-list': PT.string.isRequired,
  }).isRequired,
  list: PT.arrayOf(PT.shape({
    id: PT.number.isRequired,
    title: PT.string.isRequired,
    selected: PT.bool,
    items: PT.array.isRequired, // the structruct same as list,
  })).isRequired,
  onSelected: PT.func,
};

export default themr('Contentful-Blog', defaultTheme)(ChildListInner);
