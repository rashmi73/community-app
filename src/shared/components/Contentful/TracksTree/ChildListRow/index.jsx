/**
 * The child list row rendering.
 */
import PT from 'prop-types';
import React, { Component } from 'react';
import { themr } from 'react-css-super-themr';

import IconArrowUpSmall from 'assets/images/tc-edu/icon-arrow-up-small.svg';
import defaultTheme from './themes/default.scss';

export class ChildListRowInner extends Component {
  constructor(props) {
    super(props);

    const { selected } = props;
    this.state = {
      isOpenChild: !!selected,
    };
  }

  render() {
    const {
      theme,
      list,
      children,
      selected,
      onSelected,
    } = this.props;

    const { isOpenChild } = this.state;

    return (
      <div
        className={`${theme['child-list-row']} ${isOpenChild ? theme.expanded : ''} ${selected ? theme.selected : ''}`}
      >
        <button
          type="button"
          onClick={() => {
            if (children) {
              this.setState({ isOpenChild: !isOpenChild });
            }
            onSelected(list);
          }}
          className={theme['child-list-row-item']}
        >
          <span>{list.title}</span>
          {children && (<IconArrowUpSmall />)}
        </button>
        {isOpenChild && children}
      </div>
    );
  }
}

ChildListRowInner.defaultProps = {
  children: null,
  selected: false,
  onSelected: () => {},
  list: [],
};

ChildListRowInner.propTypes = {
  theme: PT.shape({
    'child-list-row': PT.string.isRequired,
    'child-list-row-item': PT.string.isRequired,
    expanded: PT.string.isRequired,
    selected: PT.string.isRequired,
  }).isRequired,
  list: PT.shape({
    id: PT.number.isRequired,
    title: PT.string.isRequired,
  }),
  children: PT.node,
  selected: PT.bool,
  onSelected: PT.func,
};

export default themr('Contentful-Blog', defaultTheme)(ChildListRowInner);
