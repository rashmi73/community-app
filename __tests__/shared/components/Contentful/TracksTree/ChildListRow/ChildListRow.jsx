import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import { ChildListRowInner } from 'components/Contentful/TracksTree/ChildListRow';

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();
  const MOCK_PROPS = {
    theme: {},
  };
  renderer.render(<ChildListRowInner {...MOCK_PROPS} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
