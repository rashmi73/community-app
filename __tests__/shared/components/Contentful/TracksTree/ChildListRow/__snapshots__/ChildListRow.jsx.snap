// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="undefined  "
>
  <button
    onClick={[Function]}
    type="button"
  >
    <span />
  </button>
</div>
`;
